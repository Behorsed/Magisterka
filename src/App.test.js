import { configure, shallow, mount, render, unmount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React, { useState } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import App from "./App"
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
configure({ adapter: new Adapter() });

describe('Components render without crashing', () => {
    it('renders without crashing', () => {
        shallow(<App />);
    });
});
describe('Clicking buttons in Title Screen', () => {
    it('Clicking SRT Test Button opens Simple Reaction Time test', () => {
        //arrange
        const component = mount(
            <App />
        );
        //act
        component
            .find('button.btnsimple')
            .simulate('click');
        //assert
        expect(component
            .exists('#test-simple')).toEqual(true);
    });
    it('Clicking DRT Test Button opens Discrimination Reaction Time test', () => {
        //arrange
        const component = mount(
            <App />
        );
        //act
        component
            .find('button.btndisc')
            .simulate('click');
        //assert
        expect(component
            .exists('#test-disc')).toEqual(true);
    });
});
