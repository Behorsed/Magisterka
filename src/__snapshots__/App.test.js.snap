// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App AppMatchesSnapshot 1`] = `undefined`;

exports[`AppMatchesSnapshot 1`] = `
<App>
  <div>
    <Title
      onStartDiscClick={[Function]}
      onStartSimpleClick={[Function]}
    >
      <div
        id="title"
      >
        <div
          className="simple"
        >
          <h1>
            Simple reaction time (SRT) test 
          </h1>
          <p>
            Rules: click on the circle as soon as it appears on the screen. The application will measure the time you needed to react.
          </p>
          <button
            className="btnsimple"
            onClick={[Function]}
          >
            SRT test
          </button>
        </div>
        <div
          id="disc"
        >
          <h1>
            Discrimination reaction time (DRT) test
          </h1>
          <p>
            Rules: click on the pink circle as soon as it appears on the screen. The application will measure the time you needed to react. Ignore the blue circle!
          </p>
          <button
            className="btndisc"
            onClick={[Function]}
          >
            DRT test
          </button>
        </div>
      </div>
    </Title>
  </div>
</App>
`;
